hedera-node/hedera-app/src/main/java/com/hedera/node/app/config/ConfigProviderImpl.java

```diff
@@ -47,31 +47,25 @@ public class ConfigProviderImpl implements ConfigProvider {
      */
     public ConfigProviderImpl(@NonNull final PropertySource propertySource) {
         this.propertySource = Objects.requireNonNull(propertySource, "propertySource");
-        update();
+        final Configuration config = new ConfigurationAdaptor(propertySource);
+        configuration = new AtomicReference<>(new VersionedConfigImpl(config, 0));
     }
 
     /**
      * This method must be called if a property has changed. It will update the configuration and increase the version.
      * This should happen whenever {@link GlobalDynamicProperties#reload()} is called.
      */
     public void update() {
-        try (final Locked ignored = lock.lock()) {
-            configuration = new ConfigurationAdaptor(propertySource);
-            version = version + 1;
-        }
-    }
-
-    @Override
-    public Configuration getConfiguration() {
-        try (final Locked ignored = lock.lock()) {
-            return configuration;
+        try (final var lock = updateLock.lock()) {
```

Comment:
You could also make `update()` `synchronized`, right? Or does it have an advantage to use an explicit lock here?